### PR TITLE
feat(php): add bin/console (Symfony) command

### PR DESCRIPTION
**Pull request by matthiashamacher**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature

**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**
Adding completions for Symfony bin/console

### CODE REVIEW

File: src/php/bin-console.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  generateSpec: async (tokens, executeShellCommand) => {
+    const out = await executeShellCommand("php bin/console list --format=json");
+    const subcommands = [];
+
+    try {
+      const commandDefinitions = JSON.parse(out);
+
+      commandDefinitions.commands.map((command) => {
+        if (command.name === "_complete") {
+          return;
```
Comment (fedeci) [inline src/php/bin-console.ts:13]: Map should always return some value, cau you run a filter before or combine both the map and the filter in a reduce statement?

File: src/php/bin-console.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          icon: "https://cdn.iconscout.com/icon/free/png-128/symfony-282493.png",
+          args: Object.keys(command.definition.arguments).map((argumentKey) => {
+            const argument = command.definition.arguments[argumentKey];
+
+            return {
+              name: argument.name,
+              description: argument.description,
+              isOptional: !argument.is_required,
+            };
+          }),
```
Comment (fedeci) [inline src/php/bin-console.ts:28]: nit: you can directly map through `command.definition.arguments`.
```js
command.definition.arguments.map(argument => {...})
```

File: src/php/bin-console.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          options: Object.keys(command.definition.options).map((optionKey) => {
```
Comment (fedeci) [inline src/php/bin-console.ts:29]: Same here

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks, there are a couple of nits but the spec looks awesome!

File: src/php/bin-console.ts (hunk 1 above)
Comment (matthiashamacher) [reply src/php/bin-console.ts]: Done.

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**